README.md

```diff
@@ -1,2 +1,170 @@
-# passwordless-java
-WIP
+# Passwordless Java SDK
+
+The official [Bitwarden Passwordless.dev](https://passwordless.dev/) Java library, for Java 8+ runtime.
+
+## Installation
+
+**Maven**: add to the `<dependencies>` in *pom.xml* file:
+
+```xml
+
+<dependency>
+    <groupId>com.bitwarden</groupId>
+    <artifactId>passwordless</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+</dependency>
+```
+
+**Gradle**: add to the `dependencies` in *gradle.build* file:
+
+```groovy
+implementation group: 'org.bitwarden', name: 'passwordless', version: '0.0.1-SNAPSHOT'
+```
+
+### Dependencies
+
+- [Apache HttpClient][apache-http-client] for HTTP API
+- [FasterXML jackson-databind][fasterxml-jackson-databind] for JSON (de)serialization
+
+## Getting Started
+
+Follow the [Get started guide][api-docs].
+
+### Create `PasswordlessClient` instance:
+
+```java
+import com.bitwarden.passwordless.*;
+
+import java.io.*;
+
+public class PasswordlessJavaSdkExample implements Closeable {
+
+    private final PasswordlessClient client;
+
+    public PasswordlessClientExample() {
+        PasswordlessOptions options = PasswordlessOptions.builder()
+                .apiPrivateKey("your_api_secret")
+                .build();
+
+        client = PasswordlessClientBuilder.create(options)
+                .build();
+    }
+
+    @Override
+    public void close() throws IOException {
+        client.close();
+    }
+}
+```
+
+**Note:** You need to close the underlying http client resources when you are done
+using `PasswordlessClient` with `close` method.
+
+### Register a passkey
+
+```java
+import com.bitwarden.passwordless.*;
+import com.bitwarden.passwordless.error.*;
+import com.bitwarden.passwordless.model.*;
+
+import java.io.*;
+import java.util.*;
+
+public class PasswordlessJavaSdkExample {
+
+    private final PasswordlessClient client;
+
+    // Constructor
+
+    public String getRegisterToken(String alias) throws PasswordlessApiException, IOException {
+
+        // Get existing userid from session or create a new user.
+        String userId = UUID.randomUUID().toString();
+
+        // Options to give the Api
+        RegisterToken registerToken = RegisterToken.builder()
```

Comment:
Question regarding the builder pattern, could I theoretically write the following code?
```java
RegisterToken token = RegisterToken.builder()
  .userId("myUserId")
  .build();
```

Since a username is required via the API when would I get an error?
- At compile time, does it recognize that you haven't fully given it all the data needed to build
- When `.build()` is called, does it know it's missing the required `username` field and throw?
- When the developer uses it in the `client.registerToken` and we return an API error that gets packaged up and thrown as an exception?

Just trying to understand when a developer might learn they did something wrong and how the builder pattern helps/hurts that. I know it's super common in Java and I love it for all the optional settings.

Reply:
I will answer here:
- It will not complain at compile time. I was thinking about immutables http://immutables.github.io/, which provides more options and it would fail at compile time, but initially i thought that would be an overkill, since it's simple app.
- `.build()` will fail with exception.
- Client will not be able to call the `client.registerToken`, as you need instance of the `RegisterToken` object build, which you won't be able to do without an exception if you omit required fields.
- It should be very obvious at first glance to all Java devs on which parameters are required and which are not (they are annotated).

Reply:
Awesome, thank you for that great answer!